Filtering location samples updates before syncing with server

BarAgam opened the pull request.

barmako committed:
Filtering self-location server updates.
SelfLocationUpdates will be sent to the server only if the new sample meets at least one of the following conditions:
 1. New location sample time is newer than the last sample by 20 seconds
 2. Distance delta between the the new sample and the last sent sample is more than 25 meters.

barmako committed:
Filtering self-location server updates.
SelfLocationUpdates will be sent to the server only if the new sample meets at least one of the following conditions:
 1. New location sample time is newer than the last sample by 20 seconds
 2. Distance delta between the the new sample and the last sent sample is more than 25 meters.

teamagam5 commented on domain/src/main/java/com/teamagam/gimelgimel/domain/config/Constants.java:
```diff
@@ -10,4 +10,6 @@
 
     public static final int LAYER_CACHING_RETRIES = 3;
     public static final long LAYER_CACHING_RETRIES_DELAY_MS = 2 * 1000;
+    public static final int LOCATION_CHANGE_METERS_SERVER_UPDATE_THRESHOLD = 20;
```
CHANGE_METERS -> DISTANCE_CHANGE.
In addition add units (M, MS?)

barmako committed:
1. Changed constants names
2. Increased location sampling server syncing intervals (1 min) and user location stale threshold to 5 mins
